### PR TITLE
Zap docs updates sto 7295

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
## Purpose

- Release Notes and ZAP Scanner Reference updates to reflect support for ZAP authentication and spidering 
- Updates to 1.90.1 release notes

## Preview links

- STO release notes > [Version 1.90.1](https://661028b9ae68601a0bbf3c32--harness-developer.netlify.app/release-notes/security-testing-orchestration#version-1901)
- ZAP scanner reference > [Important notes for running ZAP scans in STO](https://661028b9ae68601a0bbf3c32--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/zap/zap-scanner-reference/#important-notes-for-running-zap-scans-in-sto)
- ZAP scanner reference > [Context Name](https://661028b9ae68601a0bbf3c32--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/zap/zap-scanner-reference/#context-name)
- What's supported > [Scanner binaries used in STO container images](https://661028b9ae68601a0bbf3c32--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/security-step-settings-reference#scanner-binaries-used-in-sto-container-images) (updated BDH version)

## Merge requirements

- [x] Successful preview build.
- [x] Code owner review.
- [x] Peer review
- [x] No merge conflicts.
- [x] Release notes/new features docs: Feature/version released to at least one prod environment.

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

### CODE REVIEW

File: docs/security-testing-orchestration/sto-techref-category/zap/zap-scanner-reference.md [hunk 1]
```diff
@@ -15,12 +15,26 @@ For an example workflow, go to the [DAST app scans using Zed Attack Proxy (ZAP)
...
+  - AJAX spidering with Firefox and Selenium. Other browsers such as Chrome are not currently supported.
+  - [Script-based authentication](https://www.zaproxy.org/docs/desktop/start/features/authmethods/#scriptBased)
+  - [Form-based authentication](https://www.zaproxy.org/docs/desktop/start/features/authmethods/#formBased)
+  - [Script-based session management](https://www.zaproxy.org/docs/desktop/start/features/sessionmanagement/#sbsm) – ECMAScript / JavaScript using Nashorn engine
+    - Other languages such as Zest, Groovy, Python etc. are not currently supported.
+
+- ZAP is a highly configurable tool with many options. You should verify that your context, authentication, and other ZAP scripts work as intended before adding them to your STO pipeline.
+
+- Place your ZAP scripts in the following shared paths for your scan stage:
+  - Context scripts: `/shared/customer_artifacts/context`
```
Comment (aklaft) [inline docs/security-testing-orchestration/sto-techref-category/zap/zap-scanner-reference.md:29]: it's generally called a context file, not a script

File: docs/security-testing-orchestration/sto-techref-category/zap/zap-scanner-reference.md [hunk 2]
```diff
@@ -15,12 +15,26 @@ For an example workflow, go to the [DAST app scans using Zed Attack Proxy (ZAP)
...
+  - To verify the correct script-type subfolder, go to the [ZAP community-scripts repo](https://github.com/zaproxy/community-scripts/tree/main).
+    - Examples:
+      - `/shared/customer_artifacts/scripts/session`
+      - `/shared/customer_artifacts/scripts/authentication`
+
+- You need to [specify the full path](#context-name) to the ZAP Context script. 
 
-  * **/shared/customer_artifacts/authScript/`<artifact_file_name>`**
-  * **/shared/customer_artifacts/context/`<artifact_file_name>`**
-  * **/shared/customer_artifacts/urlFile/`<artifact_file_name>`**
```
Comment (aklaft) [inline docs/security-testing-orchestration/sto-techref-category/zap/zap-scanner-reference.md:22]: url file and hosts are still present

File: docs/security-testing-orchestration/sto-techref-category/zap/zap-scanner-reference.md (hunk 1 above)
Comment (douglas-j-bothwell) [reply docs/security-testing-orchestration/sto-techref-category/zap/zap-scanner-reference.md:29]: @aklaft Rewrote to say: "You should verify that your context file and your authentication and other scripts work as intended before adding them to your STO pipeline."

File: docs/security-testing-orchestration/sto-techref-category/zap/zap-scanner-reference.md (hunk 2 above)
Comment (douglas-j-bothwell) [reply docs/security-testing-orchestration/sto-techref-category/zap/zap-scanner-reference.md:22]: @aklaft rewrote to say: 
- Add the following shared paths (**Overview** > **Shared Paths**) to your scan stage and copy your ZAP scripts to these paths:
  - Context scripts: `/shared/customer_artifacts/context`
  - All other scripts: `/shared/customer_artifacts/scripts/<script-type>/filename`
  - For other script paths, go to the [ZAP community-scripts repo](https://github.com/zaproxy/community-scripts/tree/main).
    - Examples:
      - `/shared/customer_artifacts/scripts/session`
      - `/shared/customer_artifacts/scripts/authentication`
      - `/shared/customer_artifacts/scripts/hosts`
      - `/shared/customer_artifacts/scripts/urlFile`

**Review by brian-f-harness — APPROVED** [review]
LGTM

**Review by aklaft — APPROVED** [review]

**Comment by priteshchandaliya** [comment]
LGTM, good job @douglas-j-bothwell @aklaft

**Pull request merged**